src/sql/workbench/browser/modelComponents/tree.component.ts

```diff
@@ -76,7 +78,7 @@ export default class TreeComponent extends ComponentBase<azdata.TreeProperties>
 	}
 
 	public setDataProvider(handle: number, componentId: string, context: any): any {
-		this._dataProvider = new TreeViewDataProvider(handle, componentId, context);
+		this._dataProvider = new TreeViewDataProvider(handle, componentId, context, this._notificationService);
```

Comment:
You should use the instantiation service to create the TreeViewDataProvider instead - it'll handle injecting the services automatically (in case more are added later)

(side note - this is one reason why optional service params are not great...)

Reply:
Oh hm, I guess it's not marked as a service dependency. Oh well - in that case can you at least make the param not optional if it's going to break functionality to not have it?